Model/RatepayInvoicePaymentMethod.php

```diff
@@ -278,9 +281,88 @@ public function addMandatoryTransactionData(&$oTransaction, $oOrder)
      */
     public function isPaymentPossible()
     {
-        // no need to handle if basket amount is within range, this is checked by oxid
-        // TODO: add additional checks as soon as values are available
-        return !SessionHelper::isDateOfBirthSet() || SessionHelper::isUserOlderThan(18);
+        $oSession = Registry::getSession();
+        $oBasket = $oSession->getBasket();
+        $oBillingCountryId = $oSession->getUser()->oxuser__oxcountryid->value;
```

Comment:
This is a `string`, right? If so, please correct the notation here and in the signatures of `_areAddressesAllowed` & `_checkSettings`.